bin/test

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+main() {
+    # parse the arguments into variables
+    declare build_dir="${1:-}"
+
+    # check if composer tests are available
+    if composer run-script -l 2>/dev/null | grep -q '^\s*test\b'; then
+        # cd into the build directory and run the tests via composer
+        cd "$build_dir" && composer test
+    else # if no tests are defined...
+        # ...display a friendly reminder to define some
+        printf "\033[0;31mNo test command found! Please define a test command in the composer.json in order to run test.\033[0m\n"
+    fi
+}
+
+# fail hard
+set -eo pipefail
+# fail harder
+set -eu
+
+# execute the main function with command line arguments
+main "$@"
```

Comment:
Please move the entire logic out of that function.

Reply:
I've done it in c7c79064f5da75e4a4c220520f9583132f4c0647